### ISSUE TITLE
[TensorBoard] add_hparams outputs undesired figures

### ISSUE THREAD

**Issue by WilliamCCHuang** [pytorch/pytorch#29906]
## 🐛 Bug

The method `add_hparams` in tensorboard is amazing, however, sometimes it generates strange figures like the following figure:

<img width="1439" alt="截圖 2019-11-16 上午12 30 21" src="https://user-images.githubusercontent.com/33038038/68959176-8574b400-0808-11ea-9dcc-86c886f24bbf.png">

The second figure in the bottom row shows that there are some gaps in loss and accuracy when the x-axis is batch_size, which is unexpected. It should look like the one showing accuracy v.s. lr.

BTW, the labels of the x-axis are hard to see if they are too close to each other. It is better to show them vertically.

## To Reproduce

```
from torch.utils.tensorboard import SummaryWriter

with SummaryWriter() as w:
    for i in range(20):
        w.add_hparams(hparam_dict={'lr': 0.1 * i, 'batch_siz': 10 * i},
                      metric_dict={'hparam/accuracy': 0.9 ** i, 'hparam/loss': 0.01 * i})
    
writer.close()
```

**Comment by lanpa** [comment]
I believe this is a bug in tensorboard. Looks like the x-axis is sorted by the leading digit.

**Comment by lanpa** [comment]
https://github.com/lanpa/tensorboardX/pull/537 This should be the fix and will be ported here after unit tests are ready.
